Pod/Classes/NSAttributedString+Markdown.swift

```diff
@@ -0,0 +1,138 @@
+            if let traits = (attributes[NSFontAttributeName] as? UIFont)?.fontDescriptor().symbolicTraits {
+                let boldChange = FormattingChange.getFormattingChange(stringHasBoldEnabled, after: traits.contains(.TraitBold))
+                let italicChange = FormattingChange.getFormattingChange(stringHasItalicEnabled, after: traits.contains(.TraitItalic))
+                var formatString = ""
+                
+                switch boldChange {
+                case .Enable:
+                    formatString += "**"
+                    closingString = "**\(closingString)"
+                case .Disable:
+                    if stringHasItalicEnabled && italicChange == .Keep {
+                        formatString += "_**_"
+                        closingString = "_"
+                    } else {
+                        formatString += "**"
+                        closingString = ""
+                    }
+                case .Keep:
+                    break
+                }
+                
+                switch italicChange {
+                case .Enable:
+                    formatString += "_"
+                    closingString = "_\(closingString)"
+                case .Disable:
+                    if stringHasBoldEnabled && boldChange == .Keep {
+                        formatString = "**_**\(formatString)"
+                        closingString = "**"
+                    } else {
+                        formatString = "_\(formatString)"
+                        closingString = ""
+                    }
+                case .Keep:
+                    break
+                }
+                
+                markdownString += formatString
+                
+                stringHasBoldEnabled = traits.contains(.TraitBold)
+                stringHasItalicEnabled = traits.contains(.TraitItalic)
+            }
+            
+            let preprocessedString = (self.string as NSString).substringWithRange(range)
+            let processedString = preprocessedString.characters.reduce("") { resultString, character -> String in
+                var stringToAppend = ""
+                
+                switch character {
+                case "\\", "`", "*", "_", "{", "}", "[", "]", "(", ")", "#", "+", "-", "!":
+                    stringToAppend = "\\\(character)"
+                case "\n", "\u{2028}":
+                    stringToAppend = "\(closingString)\(character)"
+                    if !characterOnBulletedListLine && !characterOnNumberedListLine {
+                        stringToAppend += String(closingString.characters.reverse())
+                    }
+                case "1", "2", "3", "4", "5", "6", "7", "8", "9", "0":
+                    if previousCharacter == "\n" || previousCharacter == nil {
+                        openedNumberedListStarter = true
+                    }
+                    if previousCharacter == nil { numberedListIsFirstLine = true }
```

Comment:
I want to argue that point.  I think if we have any block with a small one liner it should be ok.  We do it with `guard` all the time